docker-compose.yml

```diff
@@ -1,30 +1,52 @@
-services:
-  app:
-    build: .
+x-shared-config:
+  app_common: &app_common
     container_name: ranger_ims_server
-    user: :${DAEMON_GROUP_ID:-1420}
+    # user: 1000:${DAEMON_GROUP_ID:-1420}
     environment:
       IMS_DIRECTORY: "${IMS_DIRECTORY:-File}"
       IMS_DIRECTORY_FILE: "${IMS_DIRECTORY_FILE:-/opt/ims/conf/directory.yaml}"
       IMS_DB_HOST_NAME: "${IMS_DB_HOST_NAME:-ranger_ims_database}"
       IMS_DB_DATABASE: "${IMS_DB_DATABASE:-ims}"
       IMS_DB_USER_NAME: "${IMS_DB_USER_NAME:-ims}"
       IMS_DB_PASSWORD: "${IMS_DB_PASSWORD:-ims}"
-    volumes:
-      - ./conf:/opt/ims/conf
-      - ./:/srv/ims
     ports:
      - ${IMS_SERVER_PORT:-8080}:8080
     depends_on:
       database:
         condition: service_healthy
-    command:
-      - /opt/ims/bin/ims
-      - "--log-file"
-      - "-"
-      - "--config"
-      - "/opt/ims/conf/imsd.conf"
-      - server
+
+  app_dev: &app_dev
+    build:
+      context: ./
+      dockerfile: ./Dockerfile.dev
+    <<: *app_common
+
+  app_prod: &app_prod
+    build: .
+    <<: *app_common
+
+services:
+  app:
+    <<: *app_prod
+    volumes:
+      - ./conf:/opt/ims/conf
+      - ./:/srv/ims
+    working_dir: /srv/ims
+    develop:
+      watch:
+        - action: rebuild
+          path: ./
+          target: /src/ims
+          ignore:
+            - .venv/
```

Comment:
@srabraham The watch config has options to `rebuild` or `sync` the project files. Do you know what files in the project would benefit from sync? It looks like most files require a rebuild to take effect, but we can optimize this if there are any that don't require that.

Reply:
`src/ims/` definitely covers the whole app. Any python files require the rebuild. .xhtml and .js files don't require a rebuild. Those just get served directly from disk

Reply:
Good point about the xhtml and JS files, I'll re-work the paths to handle that.